mount/mount_windows.go

```diff
@@ -99,13 +99,13 @@ func (mounter *Mounter) MountSensitive(source string, target string, fstype stri
 		getSMBMountMutex.LockKey(source)
 		defer getSMBMountMutex.UnlockKey(source)
 
-		if output, err := newSMBMapping(options[0], options[1], source); err != nil {
+		if output, err := newSMBMapping(sensitiveOptions[0], sensitiveOptions[1], source); err != nil {
```

Comment:
Nothing in the documentation indicates the order or distribution between options/sensitiveOptions is important... assuming a particular option will be in one or the other is fairly fragile. This was raised in https://github.com/kubernetes/utils/pull/138#discussion_r378635098

Reply:
given the order and distribution of options *is* important here, I would expect something more like this:

```go
allOptions := []string{}
allOptions = append(allOptions, options...)
allOptions = append(allOptions, sensitiveOptions...)
if len(allOptions) < 2 {
...
}
...
if output, err := newSMBMapping(allOptions[0], allOptions[1], source); err != nil {
...
}
```

and to document that `sensitiveOptions` are always placed after `options`

Reply:
(and don't log the content of allOptions anywhere, since it can contain sensitive options)

Reply:
Would it be better to have a map and search for specific option keys rather than assuming the options are passed in a specific order?

Reply:
iiuc, these options don't have identifying keys

Reply:
they appear to be passed in the form "user=abc", "password=abc", so we could at least try to parse and search for the correct option?